client_ext_test.go

```diff
@@ -233,3 +273,47 @@ func (a *assertPeerInterceptor) WrapStreamingHandler(next connect.StreamingHandl
 		return next(ctx, conn)
 	}
 }
+
+type assertSchemaInterceptor struct {
+	tb testing.TB
+}
+
+func (a *assertSchemaInterceptor) WrapUnary(next connect.UnaryFunc) connect.UnaryFunc {
+	return func(ctx context.Context, req connect.AnyRequest) (connect.AnyResponse, error) {
+		if !assert.NotNil(a.tb, req.Spec().Schema) {
+			return next(ctx, req)
+		}
+		methodDesc, ok := req.Spec().Schema.(protoreflect.MethodDescriptor)
+		assert.True(a.tb, ok)
+		procedure := fmt.Sprintf("/%s/%s", methodDesc.Parent().FullName(), methodDesc.Name())
+		assert.Equal(a.tb, procedure, req.Spec().Procedure)
```

Comment:
If the assertion (line 287) fails, then the next line will cause a nil dereference panic when it tries to call `methodDesc.Parent()`.
```suggestion
		if assert.True(a.tb, ok) {
			procedure := fmt.Sprintf("/%s/%s", methodDesc.Parent().FullName(), methodDesc.Name())
			assert.Equal(a.tb, procedure, req.Spec().Procedure)
		}
```